tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -223,47 +382,42 @@ public function assertProductType(string $productReference, string $productTypeN
     }
 
     /**
-     * @Then I should get error that product name is invalid
+     * @Then I should get error that product field :fieldName is invalid
```

Comment:
You should keep the original statement, we already had a discussion with @eternoendless, Gherkin language should not be specific nor contains technical information.
```suggestion
     * @Then I should get error that product :fieldName is invalid
```

Reply:
OrderFeature context has the same scenario and it becomes ambigous if i leave it. This is what I decided to do. Other fix would be to rename the OrderFeature scenario to following :
```...i should get error that product for order "[field]" is invalid```

Reply:
@PierreRambaud This behat test is describing a real end-user situation 😄 it's the usage of BO Product page. User is using a form, the UX is is front of us.

"When I enter a weak password, field 'password' is colored in red and I am prompted an error message about the password field" is valid when talking about a login process isn't it ?

Reply:
the main problem was that Order and Product context contained scenarios that matched same string. 
Finally I reverted the changes in Product context and modified scenarios in Order context.
Instead of ```Then I should get error that quantity is invalid``` I changed to ```Then I should get error that quantity is invalid for order``` , so that Gherkin understands from which Context these assertions comes from.